feature_engine/selection/shuffle_features.py

```diff
@@ -0,0 +1,213 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.metrics import roc_auc_score, mean_squared_error, r2_score
+
+from feature_engine.dataframe_checks import _is_dataframe, \
+    _check_input_matches_training_df
+
+from feature_engine.variable_manipulation import _define_variables, \
+    _find_all_variables
+
+
+class ShuffleFeatures(BaseEstimator, TransformerMixin):
+
+    """
+    
+    ShuffleFeatures reorganizes the values inside each feature, one feature 
+    at the time, from a dataframe and determines how that permutation affects
+    the performance metric of the machine learning algorithm.
+    
+    If the variables are important, a random permutation of their values will 
+    decrease dramatically any of these metrics. Contrarily, the permutation of 
+    values should have little to no effect on the model performance metric we 
+    are assessing.
+    
+
+    Parameters
+    ----------
+
+    features_to_shuffle : str or list, default=None
+        The list of variable(s) to be shuffled from the dataframe.
+        If None, the transformer will shuffle all variables in the dataset.
+        
+    estimator: object, default = DecisionTreeClassifier or DecisionTreeRegressor
+        depending on 'regression' value
+        Estimator can be a classifier or regressor
+        
+    scoring: str
+        metric used to evaluate to determine if feature needs to be kept or removed.
+
```

Comment:
can we copy from here:
https://github.com/solegalli/feature_engine/blob/e40457ce8b4baa1e146976bf1af1bbdf6eae1305/feature_engine/discretisation/decision_tree.py#L41